### PR TITLE
Adding support for tf.sigmoid_cross_entropy_with_logits and tf.where

### PR DESCRIPTION
**Pull request by cesarsouza**
Addressing #127.

### CODE REVIEW

File: TensorFlowSharp/OperationsExtras.cs [hunk 1]
```diff
@@ -460,7 +460,76 @@ public TFOutput ClipByAverageNorm (TFOutput x, TFOutput clip_norm, string operNa
...
+		/// </summary>
+		/// 
+		/// <remarks>
+		///    Measures the probability error in discrete classification tasks in which each
+		///    class is independent and not mutually exclusive.For instance, one could
+		///    perform multilabel classification where a picture can contain both an elephant
+		///    and a dog at the same time.
+		/// </remarks>
+		/// 
+		public TFOutput sigmoid_cross_entropy_with_logits (TFOutput labels, TFOutput logits, string operName = null)
```
Comment (migueldeicaza) [inline TensorFlowSharp/OperationsExtras.cs]: This does not look right, the code below references a different method, would you mind renaming this to follow the .NET conventions for method names?

File: tests/TensorFlowSharp.Tests.CSharp/MathTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		[MemberData (nameof (sigmoidCrossEntropyData))]
+		public void Should_SigmoidCrossEntropyWithLogits (double [] labels, double [] logits, double [] expected) 
+		{
+			using (var graph = new TFGraph ())
+			using (var session = new TFSession (graph)) {
+				var tlabels = graph.Placeholder (TFDataType.Double, new TFShape (2, 2));
+				var tlogits = graph.Placeholder (TFDataType.Double, new TFShape (2, 2));
+
+				TFOutput y = graph.SigmoidCrossEntropyWithLogits (tlabels, tlogits);
+
```
Comment (migueldeicaza) [inline tests/TensorFlowSharp.Tests.CSharp/MathTests.cs]: This is the call site, it looks like this might have been out of sync?

**Comment by cesarsouza** [comment]
Hopefully fixed. I did a mistake in not realizing that this PR needed MathTests.cs which was originally included in #118 and therefore should had been based on top of that commit and not on TF# master.

**Comment by cesarsouza** [comment]
Sync'ed with master, sorry for the mishap.

**Pull request merged**